### PR TITLE
feat: implement localesStore with support adding locales from externa…

### PR DESCRIPTION
**Pull request by karandashovrz**
…l packages #82

affects: @haulmont/jmix-front-generator, @haulmont/jmix-react-antd, @haulmont/jmix-react-web

### CODE REVIEW

File: example-react-app/src/i18n/i18nInit.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    locale: "en",
+    caption: "English",
+    messages: en
+  });
+
+  antdLocalesStore.addAntdLocale({
+    localeName: "en",
+    antdLocale: enUs
+  });
+  localesStore.addLocale({
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/i18n/i18nInit.ts:19]: We need a new line here

**Review by vyacheslav-pushkin — APPROVED** [review]

File: packages/jmix-front-generator/src/generators/react-typescript/app/template/src/i18n/i18nInit.ts.ejs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      caption: "Français",
+      messages: fr
+    });
+
+    antdLocalesStore.addAntdLocale({
+      localeName: 'fr',
+      antdLocale: frFr
+    })
+  <% } -%>
+  <% if (isLocaleUsed('ru')) { -%>
```
Comment (vyacheslav-pushkin) [inline packages/jmix-front-generator/src/generators/react-typescript/app/template/src/i18n/i18nInit.ts.ejs:42]: Same problem with newline here.

File: packages/jmix-front-generator/src/generators/react-typescript/app/template/src/i18n/i18nInit.ts.ejs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      caption: "Русский",
+      messages: ru
+    });
+
+    antdLocalesStore.addAntdLocale({
+      localeName: 'ru',
+      antdLocale: ruRu
+    })
+
+  <% } -%>
```
Comment (vyacheslav-pushkin) [inline packages/jmix-front-generator/src/generators/react-typescript/app/template/src/i18n/i18nInit.ts.ejs:54]: Just noticed that `cn-zh` locale is missing.

Also, let's imagine we want to add a new locale out-of-the-box. We will add a new `.json` file and add an entry to `SUPPORTED_CLIENT_LOCALES`. Ideally that should be it. How can we avoid having to put any new locale manually into this file?

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: packages/jmix-front-generator/src/generators/react-typescript/entity-editor/write.ts [hunk 4]
```diff
@@ -43,7 +22,12 @@ export const writeEditor: WriteStage<ComponentOptions, EntityEditorTemplateModel
 
   writeEditorComponent(gen, extension, templateModel);
 
-  await writeEditorMessages(projectModel, templateModel, gen, options);
+  writeComponentI18nMessages(
+    gen.fs,
+    templateModel.className,
+    options.dirShift,
+    projectModel.project?.locales
+  );
```
[the remark this answers is not in the document]
Comment (karandashovrz) [reply packages/jmix-front-generator/src/generators/react-typescript/entity-editor/write.ts:30]: This call add dynamic keys

**Review by vyacheslav-pushkin — APPROVED** [review]

**Pull request merged**